File: problems/Swift/1.两数之和.swift
/**
 * 1.两数之和 [Easy]
 * Time Complexity: O(n), Space Complexity: O(n)
 * https://leetcode-cn.com/problems/two-sum/
 */

class Solution {
  func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var hashMap: [Int: Int] = [:]
    for index in 0 ..< nums.count {
      let other = target - nums[index]
      if let otherIndex = hashMap[other] {
        return [otherIndex, index]
      }
      hashMap[nums[index]] = index
    }
    return []
  }
}

File: problems/TypeScript/1.两数之和.ts
/**
 * 1.两数之和 Easy
 * 时间复杂度：O(n), 空间复杂度：O(n)
 * https://leetcode-cn.com/problems/two-sum/
 */

function twoSum(nums: number[], target: number): number[] {
  const hashMap: { [key: number]: number } = {};
  for (let i = 0; i < nums.length; i += 1) {
    const other = target - nums[i];
    if (hashMap[other] !== undefined) {
      return [hashMap[other], i];
    }
    hashMap[nums[i]] = i;
  }
  return [];
}
